Source Code/Ticket Management System/supportgiver_menu.c

```diff
@@ -77,143 +77,206 @@ bool searchByTags(void* ticket, void* tag)
 	return found_tag;
 }
 
-void browseTickets()
+void filterByStatusAction() 
 {
-	wprintf(L"Browse Tickets\n");
-	wprintf(L"==============\n");
 	struct Vector* tickets = ticketContainer_getAll();
-	if (tickets != NULL)
+	if (tickets == NULL) {
+		wprintf(L"No tickets in the system.\n");
+		return 0;
+	}
+	else
 	{
-		printTicketsArray(tickets);
-
-		bool go_back = false;
-		while (!go_back)
+		wchar_t* ctx;
+		wchar_t status[32];
+		wprintf(L"Please input status:\n");
+		wscanf(L"%s", status);
+		wcstok(status, "\n", &ctx);
+		struct Vector* filtered_vector = vector_filter(tickets, filterByStatus, status);
+		printTicketsArray(filtered_vector);
+		system("PAUSE");
+		vector_destroy(filtered_vector);
+		vector_destroy(tickets);
```

Comment:
You own all the tickets in the `tickets` vector. The vector does not destroy its contents, so you're leaking all the `ticket`s.